README.md

```diff
@@ -1,32 +1,139 @@
+
+  validClass : 'form-field-valid',
+  // Class added to parent of each successful validation field
+
+  realTime: false,
+  // Enable real-time form control, rather than waiting until submit
+
+  onValid: function(){},
+  // Callback on submit if the form passes validation
+
+  onError: function(){},
+  // Callback on submit if the form fails validation
+
+  validators: {},
+  // Any additional custom validators
+}
+```
+
+
+## Available validators
+
+| Name | Description |
+| --- | --- |
+| `required` | Validates on the existence of a value at all. |
+| `number` | Validates on whether the value is a number. Additionally, if either a `max` or `min` attribute (or both) exist on the input, it validates on whether the number falls within that range.|
+| `email` | Validates an email if it is valid or not, using the regex defined in [this spec](https://html.spec.whatwg.org/multipage/forms.html#states-of-the-type-attribute). |
+| `creditCard` | Validates a credit cart number |
+| `equalTo[input_name]` | Returns valid if the the two fields are equal to each other. |
+| `different[input_name]` | Returns valid if the the two fields are _not_ equal to each other. |
+| `minLength[x]` | Returns valid if the value has a min-length of x. |
+| `maxLength[x]` | Returns valid if the value has a max-length of x. |
+| `minChecked[x]` | (for checkboxes) Returns valid if the number of inputs checked is greater than x. |
+| `maxChecked[x]` | (for checkboxes) Returns valid if the number of inputs checked is less than x. |
+| `minSelected[x]` | (for select-multiple) Returns valid if the number of options selected is greater than x. |
+| `maxSelected[x]` | (for select-multiple) Returns valid if the number of options selected is less than x. |
+| `regExp[validator_name]` | Validates on whether a value matches against a supplied regex expression or not. (check the [original docs](http://lab.hasanaydogdu.com/validetta/#documentation))|
+| `remote[validator_name]` | Validates using a remote validator (check the [original docs](http://lab.hasanaydogdu.com/validetta/#documentation)) |
+| `custom[validator_name]` | Validates based on a function defined in the `validators: {}` plugin option |
+
+#### A note about selects
+
+For `<input type="select">` to work with the required validator, you'll need to add a dummy option with an empty value like so: `<option disabled selected value="">Pick one...</option>`. Otherwise it will pull the first option value and assume the input is valid.
+
+## Defining validators
+
+add a `data-validates=""` attribute to your input, with a comma-separated list of validators.
+
+```
+<input name="my_email" type="email" data-validates="required,email">
+
+<input name="my_password" type="password" data-validates="required" placeholder="enter your new password">
+
+<input name="my_password_again" type="password" data-validates="required,equalTo[my_password]" placeholder="enter your new password again">
+```
+
+### custom error messages
```

Comment:
In fact, I'd make all the titles use either "Title Case" or "Sentence case"... (just minor consistency thing)